sc-tools/sc-builder/tests/units/test_gwf_translator.cpp

```diff
@@ -4,33 +4,209 @@
  * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
  */
 
-#include "builder_test.hpp"
+#include <gtest/gtest.h>
+
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <map>
+#include <memory>
+#include <algorithm>
+#include <filesystem>
 
 #include <sc-memory/sc_utils.hpp>
 
+#include "builder_test.hpp"
+
 #include "gwf_translator.hpp"
 
-#define TEST_CORRECT_GWF SC_BUILDER_KB "/test-templates/test_template_1.gwf"
-#define TEST_INCORRECT_GWF SC_BUILDER_TEST_REPOS "/kb/example.gwf"
+#include "sc_scs_tree.hpp"
+
+using GWFTranslatorTest = ScBuilderTest;
+
+#define BASE_TEST_PATH SC_BUILDER_KB "/tests-gwf-to-scs/"
+
+std::vector<char> ReadFileToBytes(std::string const & filePath)
+{
+  std::ifstream file(filePath, std::ios::binary | std::ios::ate);
+  std::streamsize size = file.tellg();
+  file.seekg(0, std::ios::beg);
+  std::vector<char> buffer(size);
+  file.read(buffer.data(), size);
+  return buffer;
+}
+
+bool CompareFiles(std::string const & filePath1, std::string const & filePath2)
+{
+  std::vector<char> file1Bytes = ReadFileToBytes(filePath1);
+  std::vector<char> file2Bytes = ReadFileToBytes(filePath2);
+  return file1Bytes == file2Bytes;
+}
+
+std::string ReadFileToString(std::string const & filePath)
+{
+  std::ifstream file(filePath);
+  std::stringstream buffer;
+  buffer << file.rdbuf();
+  return buffer.str();
+}
+
+bool CheckGWFToSCSTranslation(std::unique_ptr<ScAgentContext> & context, std::string const & fileName)
+{
+  GWFTranslator translator(*context);
+
+  std::string const & testGWFFilePath = BASE_TEST_PATH + fileName;
+  std::string const & correctSCsFilePath = testGWFFilePath + ".scs";
+
+  std::string const & testGWFText = translator.GetXMLFileContent(testGWFFilePath);
+  std::string const & testSCsText = translator.TranslateGWFToSCs(testGWFText, BASE_TEST_PATH);
+
+  std::string const & correctSCsText = ReadFileToString(correctSCsFilePath);
+
+  SCsTree testSCsTree;
+  testSCsTree.Parse(testSCsText);
+  SCsTree correctSCsTree;
+  correctSCsTree.Parse(correctSCsText);
+  return testSCsTree.Compare(correctSCsTree)->empty();
+}
+
+TEST_F(GWFTranslatorTest, InvalidPath)
+{
+  GWFTranslator translator(*m_ctx);
+
+  std::string const filePath = BASE_TEST_PATH "invalid_path.gwf";
+  EXPECT_THROW(translator.GetXMLFileContent(filePath), utils::ExceptionParseError);
+}
 
-TEST_F(ScBuilderTest, CorrectGWF)
+TEST_F(GWFTranslatorTest, EmptyFile)
 {
   GWFTranslator translator(*m_ctx);
 
-  Translator::Params translateParams;
-  translateParams.m_fileName = TEST_CORRECT_GWF;
-  translateParams.m_autoFormatInfo = true;
-  translateParams.m_outputStructure = ScAddr::Empty;
-  EXPECT_TRUE(translator.Translate(translateParams));
+  std::string const filePath = BASE_TEST_PATH "empty_file.gwf";
+  EXPECT_THROW(translator.GetXMLFileContent(filePath), utils::ExceptionParseError);
 }
 
-TEST_F(ScBuilderTest, IncorrectGWF)
+TEST_F(GWFTranslatorTest, EmptyStatic)
```

Comment:
also add test FileWithoutStatic to see what'll happen if file is not empty and it has no static sector